j5/backends/hardware/sr/v4/motor_board.py

```diff
@@ -48,6 +65,39 @@ def is_motor_board(port: ListPortInfo) -> bool:
         and port.vid == 0x0403 and port.pid == 0x6001
 
 
+class Seriallike(Protocol):
+    """
+    Something that walks like a Serial and quacks like a Serial.
+
+    This is used instead of hardcoding the Serial class to allow it to be mocked out.
+    """
+
+    def __init__(self,
```

Comment:
This isn't something that can be refactored as it has to match the upstream `serial.Serial` class.

@trickeydan please can you show me how to mark this as ignored on Code Climate? I can't see a button for it but it's likely because I don't have permission.

Reply:
If you login to code linate, which gets perms from github, you can mark this as ignored / resolved / nofix